libblkid/src/superblocks/xfs.c

```diff
@@ -118,6 +141,29 @@ static void sb_from_disk(struct xfs_super_block *from,
 	to->sb_ifree = be64_to_cpu(from->sb_ifree);
 	to->sb_fdblocks = be64_to_cpu(from->sb_fdblocks);
 	to->sb_frextents = be64_to_cpu(from->sb_frextents);
+	to->sb_uquotino = be64_to_cpu(from->sb_uquotino);
+	to->sb_gquotino = be64_to_cpu(from->sb_gquotino);
+	to->sb_qflags = be16_to_cpu(from->sb_qflags);
+	to->sb_flags = from-> sb_flags;
+	to->sb_shared_vn = from-> sb_shared_vn;
+	to->sb_inoalignmt = be32_to_cpu(from->sb_inoalignmt);
+	to->sb_unit = be32_to_cpu(from->sb_unit);
+	to->sb_width = be32_to_cpu(from->sb_width);
+	to->sb_dirblklog = from-> sb_dirblklog;
+	to->sb_logsectlog = from-> sb_logsectlog;
+	to->sb_logsectsize = be16_to_cpu(from->sb_logsectsize);
+	to->sb_logsunit = be32_to_cpu(from->sb_logsunit);
+	to->sb_features2 = be32_to_cpu(from->sb_features2);
+	to->sb_bad_features2 = be32_to_cpu(from->sb_bad_features2);
+	to->sb_features_compat = be32_to_cpu(from->sb_features_compat);
+	to->sb_features_ro_compat = be32_to_cpu(from->sb_features_ro_compat);
+	to->sb_features_incompat = be32_to_cpu(from->sb_features_incompat);
+	to->sb_features_log_incompat = be32_to_cpu(from->sb_features_log_incompat);
+	to->sb_crc = be32_to_cpu(from->sb_crc);
+	to->sb_spino_align = be32_to_cpu(from->sb_spino_align);
+	to->sb_pquotino = be64_to_cpu(from->sb_pquotino);
+	to->sb_lsn = be64_to_cpu(from->sb_lsn);
+	to->sb_rrmapino = be64_to_cpu(from->sb_rrmapino);
```

Comment:
I guess the new v5 superblock members should be read by sb_from_disk() only when `if ((sbp->sb_versionnum & 0x0f) == 5)`, right?

Reply:
In theory yes. But as the memory will be there anyways it should be fine to read it unconditionally.